src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/BoundGenerators.cs

```diff
@@ -44,9 +56,13 @@ public BoundGenerators(ImmutableArray<TypePositionInfo> elementTypeInfo, Func<Ty
                     NativeReturnMarshaller = generator;
                     foundNativeRetMarshaller = true;
                 }
-                if (!argType.IsManagedReturnPosition && !argType.IsNativeReturnPosition)
+                if (argType.ManagedIndex is not (TypePositionInfo.UnsetIndex or TypePositionInfo.ReturnIndex or TypePositionInfo.ExceptionIndex))
```

Comment:
Let's comment on what this and the branch are doing a bit more. They seem very very close but one is computed using some expression matching whereas the below is using some boolean logic. Is there a reason? I assume one is for unmanaged => managed and the other managed => managed judging by the "paramMarshallers" collections.